LLVM incorrectly assumes `long double` is the same type as `_Float128` on ppc64le

```
/build/source/llvm/llvm/lib/Analysis/ConstantFolding.cpp:2129:34: error: invalid conversion from '_Float128 (*)(_Float128) noexcept' to 'long double (*)(long double)' [-fpermissive]
 2129 |         return ConstantFoldFP128(logf128, Op->getValueAPF(), Ty);
      |                                  ^~~~~~~
      |                                  |
      |                                  _Float128 (*)(_Float128) noexcept
/build/source/llvm/llvm/lib/Analysis/ConstantFolding.cpp:1787:43: note:   initializing argument 1 of 'llvm::Constant* {anonymous}::ConstantFoldFP128(long double (*)(long double), const llvm::APFloat&, llvm::Type*)'
 1787 | Constant *ConstantFoldFP128(long double (*NativeFP)(long double),
      |                             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
```

`ConstantFoldFP128`'s definition:
https://github.com/llvm/llvm-project/blob/599f8e1120db8c8c7a5e42134ea52650e8fb0f1c/llvm/lib/Analysis/ConstantFolding.cpp#L1786-L1798

Where it's called with mismatched types:
https://github.com/llvm/llvm-project/blob/599f8e1120db8c8c7a5e42134ea52650e8fb0f1c/llvm/lib/Analysis/ConstantFolding.cpp#L2129

--- a/llvm/lib/Analysis/ConstantFolding.cpp
+++ b/llvm/lib/Analysis/ConstantFolding.cpp
@@ -1784,8 +1784,8 @@ Constant *ConstantFoldFP(double (*NativeFP)(double), const APFloat &V,
 }
 
 #if defined(HAS_IEE754_FLOAT128) && defined(HAS_LOGF128)
-Constant *ConstantFoldFP128(long double (*NativeFP)(long double),
-                            const APFloat &V, Type *Ty) {
+Constant *ConstantFoldFP128(float128 (*NativeFP)(float128), const APFloat &V,
+                            Type *Ty) {
   llvm_fenv_clearexcept();
   float128 Result = NativeFP(V.convertToQuad());
   if (llvm_fenv_testexcept()) {

